spec/models/shared/shared_votes_counter.rb

```diff
@@ -0,0 +1,77 @@
+RSpec.shared_examples "a votable" do |described_class_factory|
+  subject { create(described_class_factory) }
+
+  context "votes" do
+    it { is_expected.to be_a described_class }
+    it { expect(subject.votes.size).to eq 5 }
+    it { expect(subject.votes.first).to be_instance_of Vote }
+  end
+
+  describe "has_many :votes" do
+    it { is_expected.to have_many(:votes) }
+  end
+
+  describe ".top" do
+    let(:resource) { described_class.to_s.downcase!.to_sym }
+
+    let(:high_vote_resource) { create(resource) }
+    before { 6.times { create(:vote, value: 1, voteable: high_vote_resource) } }
+
+    let(:low_vote_resource) { create(resource) }
+    before { 4.times { create(:vote, value: -1, voteable: low_vote_resource) } }
+
+    before { 5.times { create(described_class_factory) } }
+
+    subject { described_class.top(2) }
+
+    it { expect(subject.length).to be(2) }
+    it { is_expected.to include(high_vote_resource) }
+    it { is_expected.not_to include(low_vote_resource) }
+  end
```

Comment:
Nice. I shall make it that more specific. As for testing the `.top` for the default number of entries, I will have to generate more than 10 of the resources. :100: